programs/server/Server.cpp

```diff
@@ -613,6 +613,49 @@ static void sanityChecks(Server & server)
     }
 }
 
+void loadStartupScripts(const Poco::Util::AbstractConfiguration & config, ContextMutablePtr context, Poco::Logger * log)
+{
+    try
+    {
+        Poco::Util::AbstractConfiguration::Keys keys;
+        config.keys("startup_scripts", keys);
+
+        SetResultDetailsFunc callback;
+        for (const auto & key : keys)
+        {
+            std::string full_prefix = "startup_scripts." + key;
+
+            if (config.has(full_prefix + ".condition"))
+            {
+                auto condition = config.getString(full_prefix + ".condition");
+                auto condition_read_buffer = ReadBufferFromString(condition);
+                auto condition_write_buffer = WriteBufferFromOwnString();
+
+                LOG_DEBUG(log, "Checking startup query condition `{}`", condition);
+                executeQuery(condition_read_buffer, condition_write_buffer, true, context, callback, QueryFlags{ .internal = true }, std::nullopt, {});
+
+                auto result = condition_write_buffer.str();
+
+                if (result != "1\n" && result != "true\n")
```

Comment:
It will be tricky to debug if unexpected result is returned (several rows or another non-null value).
Let's also check that if it's not single row `0` or `false` then print `LOG_WARNING` and maybe even add to `system.warnings` to display a message about misconfiguration to user.

https://github.com/ClickHouse/ClickHouse/blob/a197a530d67d175e4cf128b88a6308c81da64ce4/src/Interpreters/SystemLog.cpp#L334

Reply:
Also adding documentation for the feature with the description of this details and some examples could be helpful.